src/gwc/src/main/java/org/geoserver/gwc/layer/CatalogStyleChangeListener.java

```diff
@@ -78,17 +78,24 @@ public void handleModifyEvent(CatalogModifyEvent event) throws CatalogException
             return;
         }
         final List<String> propertyNames = event.getPropertyNames();
-        if (!propertyNames.contains("name")) {
+        if (!propertyNames.contains("name") && !propertyNames.contains("workspace")) {
```

Comment:
Does this catch both moving to a new workspace and the workspace's name being changed?